src/services/ratingService.js

```diff
@@ -0,0 +1,118 @@
+    }
+
+    static async updateRating(id, rating){
+        try{
+            let updatedRating = await Rating.update(rating, {
+                returning: true,
+                where: id
+            })
+
+            return updatedRating;
+        }catch(error){
+            throw error;
+        }
+    }
+    //[sequelize.json('quantity.rate'),'quantity'] 
+
+    static async computeAverage(user){
+        try{
+            let allRatings = await RatingService.getRatings({ user });
+
+            //Compute Average of all ratings of a user
+            let average_rating = computeAverage(allRatings);
+
+            //Update the average ratings Table
+            await RatingService.updateAverage({user},average_rating);
+
+        }catch(error){
+            throw error;
+        }
+    }
+
+    static async getAverage(param){
+        try{
+            let average = await AverageRatings.findAll({where: param});
+            return average;
+
+        }catch(error){
+            console.log(error);
+            throw error;
+        }
+    }
+
+    static async updateAverage(user, rating){
+        try{
+            let found_average = await AverageRatings.findAll({where: user});
+
+            if(found_average.length != 0){
+                let average = await AverageRatings.update(rating, {
+                    returning: true,
+                    where: user
+                });
+
+                return average;
+            }
+            rating.user = user.user;
+            rating.submitter = 1; //Making Super LF default submitter for now
+            let average =  await AverageRatings.create(rating);
+
+            return average;
+
+        }catch(error){
+            throw error;
+        }
+    }
+}
+
+
+export default RatingService;
+
+
+ // let userRatings = await Rating.findAll({
+            //    // where: user,
+            //     attributes: ['id', 'user',
+            //     // [sequelize.json('quality.rate'),'quality'],
+            //     // [sequelize.json('quantity.rate'),'quantity'],
+
+            //     [sequelize.fn('count', sequelize.col('id')), 'count'], 
+            //     // [sequelize.json('initiative.rate'),'initiative'],
+            //     // [sequelize.json('communication.rate'),'communication'],
+            //     // [sequelize.json('professionalism.rate'),'professionalism']
+            // ],
+            // //group: ['id'],
+            // raw: true
+            // });
+              //[sequelize.fn('count', sequelize.json('communication.rate'),'communication'), 'communications'],
+
+
+              //[sequelize.json('quality.rate'),'quality'],
+                // [sequelize.json('quantity.rate'),'quantity'],
+                // [sequelize.json('initiative.rate'),'initiative'],
+                // [sequelize.json('communication.rate'),'communication'],
+                // [sequelize.json('professionalism.rate'),'professionalism'],
+                // [sequelize.json('integration.rate'),'integration']
```

Comment:
Can you delete these comments if they are unnecessary

Reply:
Will delete the comments. About an engineer being  graded twice, According to Denis an LF can rate an engineer even for 5 weeks at once. So no check is needed.

Reply:
@willywax cool, but then how will we link grades to a week? should we add it to the ratings table

Reply:
For the LF to add the week

Reply:
@drayzii Nice observation. But according to the PO we don't provide ratings weekly. So it's possible to provide rating to an Engineer even 4 weeks in a single day.